src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorLanguageServer.cs

```diff
@@ -72,7 +72,26 @@ public static Task<ILanguageServer> CreateAsync(Stream input, Stream output, Tra
                     .WithHandler<RazorSemanticTokenLegendEndpoint>()
                     .WithServices(services =>
                     {
-                        services.AddSingleton<FilePathNormalizer>();
+                        var filePathNormalizer = new FilePathNormalizer();
+                        services.AddSingleton<FilePathNormalizer>(filePathNormalizer);
+
+                        var foregroundDispatcher = new DefaultForegroundDispatcher();
+                        services.AddSingleton<ForegroundDispatcher>(foregroundDispatcher);
+
+                        var generatedDocumentPublisher = new DefaultGeneratedDocumentPublisher(foregroundDispatcher, new Lazy<OmniSharp.Extensions.LanguageServer.Protocol.Server.ILanguageServer>(() => server));
```

Comment:
This all got moved up because the `FilePathNormalizer` had to be instantiated right?

Reply:
Yup. And all the other types we need to initiate in order.